tools/metrics/gauges_rds_db_manual_snapshots.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"code.cloudfoundry.org/lager"
+	m "github.com/alphagov/paas-cf/tools/metrics/pkg/metrics"
+	"github.com/alphagov/paas-cf/tools/metrics/pkg/rds"
+	"github.com/alphagov/paas-cf/tools/metrics/pkg/servicequotas"
+	"github.com/aws/aws-sdk-go/aws"
+	awsrds "github.com/aws/aws-sdk-go/service/rds"
+	awsservicequotas "github.com/aws/aws-sdk-go/service/servicequotas"
+)
+
+func RDSDBManualSnapshotsGauge(
+	logger lager.Logger,
+	rds *rds.RDSService,
+	serviceQuotas *servicequotas.ServiceQuotas,
+	interval time.Duration,
+) m.MetricReadCloser {
+	return m.NewMetricPoller(interval, func(w m.MetricWriter) error {
+		lsess := logger.Session("rds-db-manual-snapshots-gauge")
+		count, err := countRDSManualSnapshots(rds)
+		if err != nil {
+			lsess.Error("count-manual-snapshots", err)
+			return err
+		}
+		quota, err := getRDSManualSnapshotQuota(serviceQuotas)
+		if err != nil {
+			lsess.Error("get-db-quota", err)
+			return err
+		}
+		metrics := []m.Metric{
+			{
+				Kind:  "gauge",
+				Name:  "aws.rds.manual.snapshot.count",
+				Value: float64(count),
+				Unit:  "count",
+			},
+			{
+				Kind:  "gauge",
+				Name:  "aws.rds.manual.snapshot.quota.count",
+				Value: quota,
+				Unit:  "count",
+			},
+		}
+
+		return w.WriteMetrics(metrics)
+	})
+}
+
+func countRDSManualSnapshots(service *rds.RDSService) (int, error) {
+	snapshots, err := service.Client.DescribeDBSnapshots(
+		&awsrds.DescribeDBSnapshotsInput{
+			SnapshotType: aws.String("manual"),
+		})
+
+	if err != nil {
+		return 0, err
+	}
+
+	return len(snapshots.DBSnapshots), nil
+}
+
+func getRDSManualSnapshotQuota(service *servicequotas.ServiceQuotas) (float64, error) {
```

Comment:
Is there a reason for using `float64` instead of `int64`?

Reply:
The end result needs to be a float due to it ending up in a prometheus metric.

```
			{
				Kind:  "gauge",
				Name:  "aws.rds.manual.snapshot.count",
				Value: float64(count),
				Unit:  "count",
			},
			{
				Kind:  "gauge",
				Name:  "aws.rds.manual.snapshot.quota.count",
				Value: quota,
				Unit:  "count"
                         }
```
I'm happy to change the function to return an int and then convert to a float as we do with the count. 

Want me to make the change?

Reply:
Oh I see - I was curious about why it was different. We don't need to change it if it makes no difference 👍

Reply:
I agree it's inconsistent. It does need to end up in a float.

Like I say happy to change if you want. Up to you.